backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/database/service/SpecificationDatabaseService.kt

```diff
@@ -13,17 +14,32 @@ import javax.transaction.Transactional
 @Service
 class SpecificationDatabaseService constructor(
     private val specificationRepository: SpecificationRepository,
-    private val entityManager: EntityManager
+    private val entityManager: EntityManager,
+    private val specificationEntityMapper: SpecificationEntityMapper
 ) : SpecificationPersistenceService {
 
     override fun save(specification: Specification) {
-        specificationRepository.save(SpecificationEntity.fromDomain(specification))
+        val specificationEntity = specificationEntityMapper.fromDomain(specification)
+
+        if (specificationRepository.existsByTitle(specificationEntity.title)) {
```

Comment:
Ideally, we could use a UUID for this instead of the name.

Reply:
But when you create a new Specification and want to check if some Specification with the same name already exists? That's the use case here.

Reply:
I'm not sure if we should prevent two specs from being called by the same name. There may be a legitimate reason for that.